tooling/nargo_cli/tests/execute.rs

```diff
@@ -341,19 +360,50 @@ mod tests {
             insta::assert_json_snapshot!(snapshot_name, artifact, {
                 ".noir_version" => "[noir_version]",
                 ".hash" => "[hash]",
-                ".file_map.**.path" => insta::dynamic_redaction(|value, _path| {
-                    // Some paths are absolute: clear those out.
-                    let value = value.as_str().expect("Expected a string value in a path entry");
-                    if value.starts_with("/") {
-                        String::new()
-                    } else {
-                        value.to_string()
-                    }
-                }),
+                ".file_map.**.path" => file_map_path_redaction(),
+            })
+        })
+    }
+
+    fn check_contract_artifact(
+        prefix: &'static str,
+        test_program_dir: &Path,
+        target_dir: &PathBuf,
+        force_brillig: ForceBrillig,
+        inliner: Inliner,
+    ) {
+        let artifact_filename =
+            find_program_artifact_in_dir(target_dir).expect("Expected an artifact to exist");
+
+        let artifact_file = fs::File::open(&artifact_filename).unwrap();
+        let artifact: ContractArtifact = serde_json::from_reader(artifact_file).unwrap();
+
+        let _ = fs::remove_dir_all(target_dir);
```

Comment:
Is this silently ignoring failures?

Reply:
Yes, but I'm not sure we should fail the test because of a missing cleanup. Or yes?

Reply:
Yes, I think so: missing cleanup could cause knock-on effects that we could have trouble debugging otherwise

Reply:
Done!